EchoPro/survey.py

```diff
@@ -36,741 +23,440 @@ class Survey:
-            self.nasc_df = load_nasc_df(self)
+        ### Format dictionary that will parameterize the `config` class attribute
+        # Join the initialization and survey year parameters into a single dictionary
+        # Pass 'full_params' to the class instance
+        return {**init_config_params, **survey_year_config_params}
 
-    def compute_transect_results(
-        self, selected_transects: Optional[List] = None
-    ) -> None:
+    def load_survey_data( self ):
         """
-        Constructs ``self.bio_calc.transect_results_gdf``,
-        ``self.bio_calc.transect_results_male_gdf``, and
-        ``self.bio_calc.transect_results_female_gdf``, which are
-        GeoDataFrames that contain variables over the transect
-        points (e.g. abundance, biomass).
-
-        Parameters
-        ----------
-        selected_transects : list or None
-            The subset of transects used in the calculations
-        """
-
-        self.bio_calc = None
-        self.bio_calc = ComputeTransectVariables(self)
-        self.bio_calc.get_transect_results_gdf(selected_transects)
-
-        # create Dataset containing useful distributions and variables over length and age
-        self.bio_calc.bin_ds = generate_bin_ds(self)
-
-        # add NASC_adult to transect_results_gdf (needs to occur after generate_bin_ds)
-        self.bio_calc.set_adult_NASC()
-
-        # calculate Jolly-Hampton CV
-        self.bio_calc.jollyhampton_cv = self.run_cv_analysis(kriged_data=False)
-
-        # compute abundance and biomass for each length and age bin
-        self.compute_length_age_variables(data="transect")
+        Loads the biological, NASC, and stratification
+        data using parameters obtained from the configuration
+        files. This will generate data attributes associated with the tags
+        defined in both the configuration yml files and the reference CONFIG_MAP
+        and LAYER_NAME_MAP dictionaries.
+        """
+
+        ### Check whether data files defined from the configuration file exists
+        # Generate flat JSON table comprising all configuration parameter names
+        flat_configuration_table = pd.json_normalize(self.config).filter(regex="filename")
+
+        # Parse the flattened configuration table to identify data file names and paths
+        parsed_filenames = flat_configuration_table.values.flatten()
+
+        # Append the root directory to the parsed data file names
+        data_paths = [Path(self.config['data_root_dir'] + '/' + file) for file in parsed_filenames]
+
+        # Evaluate whether either file is missing
+        data_existence = [path.exists() for path in data_paths]
+
+        # Assign the existence status to each configuration file for error evaluation
+        # Error evaluation and print message (if applicable)
+        if not all(data_existence):
+            missing_data = [ files for files, exists in zip( parsed_filenames, data_existence ) if not exists ]
```

Comment:
You can index the list directly:
```suggestion
            missing_data = parsed_filenames[data_existence]
```